### PR TITLE
Fix deck resubmission process

### PR DESCRIPTION
**Pull request by AlphaKretin**
## Description

Bar players from clicking the button to register when already registered
On deck resubmission through DM, disable validation on previous deck

## Checklist

- [x] I am following the [contributing guidelines](https://github.com/DawnbrandBots/emcee-tournament-bot/blob/master/.github/CONTRIBUTING.md).
- [ ] I have updated any relevant [user guides](https://github.com/DawnbrandBots/emcee-tournament-bot/blob/master/guides).
- [ ] I have updated any relevant [documentation](https://github.com/DawnbrandBots/emcee-tournament-bot/blob/master/docs).
- [ ] I have updated the [changelog](https://github.com/DawnbrandBots/emcee-tournament-bot/blob/master/changelog.md), or this change is not user-facing.

### CODE REVIEW

**Comment by AlphaKretin** [comment]
Tested before merge

File: src/slash/deck.ts [hunk 1]
```diff
@@ -107,7 +107,18 @@ export class AcceptButtonHandler implements ButtonClickHandler {
...
+		const deck = await ManualDeckSubmission.findOneOrFail({
+			where: {
+				discordId: interaction.user.id,
+				tournamentId
+			}
+		});
+		if (deck.message !== sourceMessageId) {
+			await interaction.reply(`This deck is outdated, the player has submitted a newer one!`);
+			return;
+		}
```
Comment (kevinlul) [inline src/slash/deck.ts:121]: Post-merge, like what I did, prepend "OUTDATED" and remove buttons

**Comment by kevinlul** [comment]
coverage successfully tanked to 40%

**Comment by AlphaKretin** [comment]
Is a test suite for slash commands reasonable when we have more time?

**Comment by kevinlul** [comment]
I think so, with testcontainers or the in-memory SQLite, we can do some decent tests. I'm not happy with how the mocking is done in Bastion currently though, so more research is warranted.

**Review by kevinlul — APPROVED** [review]
🚀

**Pull request merged**